skxray/roi.py

```diff
@@ -106,218 +104,122 @@ def rectangles(num_rois, roi_data, image_shape):
-    # discard the indices greater than number of rings
-    labels_grid[labels_grid > num_rings] = 0
-
-    return labels_grid.reshape(image_shape)
+    print(np.unique(label_array))
+    # Even elements of label_array are in the space between rings.
+    label_array = (np.where(label_array % 2 != 0, label_array, 0) + 1) // 2
+    print(np.unique(label_array))
+    return label_array.reshape(shape)
 
 
-def rings_edges(num_rings, first_r, delta_r):
+def ring_edges(inner_radius, width, spacing=0, num_rings=None):
     """
-    This function will provide the edge values of the rings when
-    there is a no step value between each ring
+    Calculate the inner and outer radius of a set of rings.
+
+    The number of rings, their widths, and any spacing between rings can be
+    specified. They can be uniform of varied.
 
     Parameters
     ----------
-    num_rings : int
-        number of rings
-
-    first_r : float
-        radius of the first  ring
-
-    delta_r : float
-        thickness of the ring
-
-    Returns
-    -------
-    ring_vals : array
-        edge values of each ring
-
-    ring_edges : array
-        edge values of each ring
-        shape is (num_rings, 2)
-    """
+    inner_radius : float
+        inner radius of the inner-most ring
 
-    # last ring edge value
-    last_r = first_r + num_rings*(delta_r)
+    width : float or list of floats
+        ring thickness
+        If a float, all rings will have the same thickness.
 
-    # edges of all the rings
-    ring_edges = np.linspace(first_r, last_r, num=(num_rings+1))
+    spacing : float or list of floats, optional
+        margin between rings, 0 by default
+        If a float, all rings will have the same spacing. If a list,
+        the length of the list must be one less than the number of
+        rings.
 
-    return ring_edges
-
-
-def rings_step(image_shape, calibrated_center, num_rings, first_r, delta_r,
-               *step_r):
-    """
-    This will provide the indices of the required rings,
-    find the bin edges of the rings, and count the number
-    of pixels in each ring, and pixels indices for the
-    required rings when there is a step value between rings.
-    Step value can be same or different steps between
-    each ring.
-
-    num_rings : int
+    num_rings : int, optional
         number of rings
-
-    first_r : float
-        radius of the first ring
-
-    delta_r : float
-        thickness of the ring
-
-    calibarted_center : tuple
-        defining the center of the image (column value, row value) (mm)
-
-    step_r : tuple
-        step value for the next ring from the end of the previous
-        ring.
-        same step - same step values between rings (one value)
-        different steps - different step value between rings (provide
-        step value for each ring eg: 6 rings provide 5 step values)
+        required if width and spacing are not lists and number
+        cannot thereby be inferred
```

Comment:
Consistency is checked, and an error is raised if they disagree. I elaborated in the docstring to make this clear.